spec.html

```diff
@@ -31129,7 +31129,8 @@ <h1>RegExp.prototype [ @@matchAll ] ( _string_ )</h1>
           1. If Type(_R_) is not Object, throw a *TypeError* exception.
           1. Let _S_ be ? ToString(_string_).
           1. Let _C_ be ? SpeciesConstructor(_R_, %RegExp%).
-          1. Let _flags_ be ? ToString(? Get(_R_, `"flags"`)).
+          1. Let _flagsValue_ be ? Get(_R_, `"flags"`).
```

Comment:
Do you think that `@@split` suffers from a similar issue?

```js
var str = 'a\u{1d11e}b';
var pattern = /\u{1d11e}/u;
Object.defineProperty(pattern, 'flags', { value: undefined, configurable: true });
str.split(pattern);
// SyntaxError: Invalid flags supplied to RegExp constructor 'undefinedy'
```

`undefinedy` 🤣

Reply:
Yes, that also seems like the same spec bug to me.

I'd be happy to pursue changing this in the other methods too, if folks feel that's viable.

Reply:
If we're going to do this in multiple places, should we have something like a `ToStringOrEmpty` operation to handle `null` or `undefined`?

BTW: `null` is an issue too, both for matchAll and split:

```js
var str = 'a\u{1d11e}b';
var pattern = /\u{1d11e}/u;
Object.defineProperty(pattern, 'flags', { value: null, configurable: true });
str.split(pattern);
// SyntaxError: Invalid flags supplied to RegExp constructor 'nully'
```